plugins/CoreAdminHome/CustomLogo.php

```diff
@@ -163,6 +163,16 @@ protected static function rewritePath($path)
         return SettingsPiwik::rewriteMiscUserPathWithInstanceId($path);
     }
 
+    public static function hasUserLogo()
+    {
+        return file_exists(Filesystem::getPathToPiwikRoot() . '/' . CustomLogo::getPathUserLogo());
```

Comment:
good point, I didn't notice that. Consistency is always good. We usually always use `PIWIK_INCLUDE_PATH ` but not sure why `Filesystem::getPathToPiwikRoot()` was used there is there's no difference and the constant should be better. I'm ok with `Filesystem::getPathToPiwikRoot()` in this case. To remove the duplicated code though maybe there could be a method  like

```php
private function getRootPath()
{
    return Filesystem::getPathToPiwikRoot() . '/';
}
// or 
private function getAbsolutePath($reference)
{
    return Filesystem::getPathToPiwikRoot() . '/' . $reference;
}
```